### DISCUSSION TITLE
ESM  support?

### DISCUSSION THREAD

**Discussion by ziimakc** [remix-run/remix#7419] [Q&A]
Does project supports esm? If so, what do I need to configure to make `remix dev` work?

```
/xxx/node_modules/.pnpm/@remix-run+serve@1.19.3/node_modules/@remix-run/serve/dist/cli.js:38
let build = require(buildPath);
            ^
Error [ERR_REQUIRE_ESM]: require() of ES Module /xxx/build/index.js from /xxx/node_modules/.pnpm/@remix-run+serve@1.19.3/node_modules/@remix-run/serve/dist/cli.js not supported.
Instead change the require of index.js in /xxx/node_modules/.pnpm/@remix-run+serve@1.19.3/node_modules/@remix-run/serve/dist/cli.js to a dynamic import() which is available in all CommonJS modules.
    at Object.<anonymous> (/xxx/node_modules/.pnpm/@remix-run+serve@1.19.3/node_modules/@remix-run/serve/dist/cli.js:38:13)
```

**Comment by brookslybrand** [comment]
Hi @ZiiMakc, yes you can choose whether you want the server build to be cjs or esm: https://remix.run/docs/en/main/file-conventions/remix-config#servermoduleformat

**Comment by andrewsolis** [comment]
Hello,

I've tried setting the `  serverModuleFormat: "esm"` inside the remix.config.js file but still get an error.

![clierror](https://github.com/remix-run/remix/assets/5217846/21969638-3d9c-45ad-a569-061e248bfacc)

I'm running **Windows 10** and just followed the tutorial [here](https://remix.run/docs/en/1.19.3/tutorials/jokes) but received this error when attempting to run my sample project using `npm start` after building.

I'm using remix version 1.19.3 as well. 

I've noticed a couple other workarounds across github but none seemed to work. Any other suggestions would be helpful.

**Reply by brookslybrand** [reply]
Have you tried using `cjs` instead?

If that doesn't work, are you able share a repo with the issue?

**Reply by andrewsolis** [reply]
I tried using cjs as well but it gave a similar error. 
I moved over to Mac OS and it seems to work there.
